metrics/web/lib/common/presentation/state/projects_notifier.dart

```diff
@@ -0,0 +1,96 @@
+import 'dart:async';
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/services.dart';
+import 'package:metrics/dashboard/domain/usecases/receive_project_updates.dart';
+import 'package:metrics_core/metrics_core.dart';
+
+/// The [ChangeNotifier] that holds the projects state.
+///
+/// Stores the [Project]s data.
+class ProjectsNotifier extends ChangeNotifier {
+
+  /// Creates the projects store.
+  ///
+  /// The provided use cases should not be null.
+  ProjectsNotifier(
+    this._receiveProjectsUpdates,
+  ) : assert(
+          _receiveProjectsUpdates != null,
+          'The use case should not be null',
+        );
+
+  /// Provides an ability to receive project updates.
+  final ReceiveProjectUpdates _receiveProjectsUpdates;
```

Comment:
The `common` package should not depend on any other modules like `dashboard`. We, probably, should move the  `ReceiveProjectUpdates ` class to the common package and create a separate repository for it.